### DISCUSSION TITLE
Clarify the use of ^| in rules please

### DISCUSSION THREAD

**Discussion by timkgh** [AdguardTeam/AdGuardHome#3768] [Q&A]
In your official list, some entries use just `^` at the end of the rule, while others at the end of the file use `^|`.

https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt

I don't _think_ it matters, or is there something subtle why some rules use `^|`?

cc @ameshkov @zelo72

**Comment by ghost** [comment]
I had also searched for documentation on this combination but found nothing.

**Reply by ameshkov** [reply]
Well, to be honest, `|` is simply redundant in the case of DNS filtering (it means that this is the "end of string" here).

However, if the same list is used in a traditional ad blocker, it makes sense and means "only match the homepage, not all other pages".

**Comment by timkgh** [comment]
For DNS rules it seems that `^` and `|` at the end are interchangeable and using both of them is not needed, it's redundant. Using `^` might be preferred as it's more explicit, it marks the end of the domain name.
